openedx/core/djangoapps/credit/api.py

```diff
@@ -322,6 +322,93 @@ def get_credit_requests_for_user(username):
     return CreditRequest.credit_requests_for_user(username)
 
 
+def get_credit_requirement_status(course_key, username):
+    """ Retrieve the user's status for each credit requirement in the course.
+
+    Args:
+        course_key (CourseKey): The identifier for course
+        username (str): The identifier of the user
+
+    Example:
+        >>> get_credit_requirement_status("course-v1-edX-DemoX-1T2015", "john")
+
+                [
+                    {
+                        "namespace": "reverification",
+                        "name": "i4x://edX/DemoX/edx-reverification-block/assessment_uuid",
+                        "criteria": {},
+                        "status": "satisfied",
+                    },
+                    {
+                        "namespace": "reverification",
+                        "name": "i4x://edX/DemoX/edx-reverification-block/assessment_uuid",
+                        "criteria": {},
+                        "status": "Not satisfied",
+                    },
+                    {
+                        "namespace": "proctored_exam",
+                        "name": "i4x://edX/DemoX/proctoring-block/final_uuid",
+                        "criteria": {},
+                        "status": "error",
+                    },
+                    {
+                        "namespace": "grade",
+                        "name": "i4x://edX/DemoX/proctoring-block/final_uuid",
+                        "criteria": {"min_grade": 0.8},
+                        "status": None,
+                    },
+                ]
+
+    Returns:
+        list of requirement statuses
+    """
+    requirements = CreditRequirement.get_course_requirements(course_key)
+    requirement_list = [requirement.id for requirement in requirements]
+    requirement_statuses = CreditRequirementStatus.get_statuses(requirement_list, username)
+    statuses = []
+    for requirement in requirements:
+        status = None
+        status_date = None
+        for requirement_status in requirement_statuses:
```

Comment:
You might consider making a dict from the statuses to avoid the nested for loops:

```python
requirement_statuses = CreditRequirementStatus.get_statuses(requirement_list, username)
requirement_statuses = dict((o.requirement, o) for o in requirement_statuses)

for requirement in requirements:
  ...
  requirement_status = requirement_statuses.get(requirement)
  ...
```